compiler/ilgen/IlValue.cpp

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2017, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ ******************************************************************************/
+
+
+
+#include "compile/Compilation.hpp"
+#include "il/Block.hpp"
+#include "il/Node.hpp"
+#include "il/Node_inlines.hpp"
+#include "il/Symbol.hpp"
+#include "il/symbol/AutomaticSymbol.hpp"
+#include "il/SymbolReference.hpp"
+#include "ilgen/IlValue.hpp" // must follow include for compile/Compilation.hpp for TR_Memory
+#include "ilgen/MethodBuilder.hpp"
+
+
+OMR::IlValue::IlValue(TR::Node *node, TR::TreeTop *treeTop, TR::Block *block, TR::MethodBuilder *methodBuilder)
+   : _id(methodBuilder->getNextValueID()),
+     _nodeThatComputesValue(node),
+     _treeTopThatAnchorsValue(treeTop),
+     _blockThatComputesValue(block),
+     _methodBuilder(methodBuilder),
+     _symRefThatCanBeUsedInOtherBlocks(0)
+   {
+   }
+
+TR::DataType
+OMR::IlValue::getDataType()
+   {
+   return _nodeThatComputesValue->getDataType();
+   }
+
+void
+OMR::IlValue::storeToAuto()
+   {
+   if (_symRefThatCanBeUsedInOtherBlocks == NULL)
+      {
+      TR::Compilation *comp = TR::comp();
+
+      // first use from another block, need to create symref and insert store tree where node  was computed
+      TR::SymbolReference *symRef = comp->getSymRefTab()->createTemporary(_methodBuilder->methodSymbol(), _nodeThatComputesValue->getDataType());
+      symRef->getSymbol()->setNotCollected();
+      char *name = (char *) comp->trMemory()->allocateHeapMemory(8 * sizeof(char));
+      sprintf(name, "_T%-5d", symRef->getCPIndex());
```

Comment:
You're correct. If we managed to create a million symbol references in a single compilation, it would overflow. I suppose a bug that permitted a garbage value in there could also overflow.

This problem pre-existed my change, but I'll add a commit here to address it. Thanks for  pointing it out @fjeremic .